### PR TITLE
Add guard clause if date missing on day_of_month calc

### PR DESCRIPTION
**Pull request by fergmastaflex**
@mattnichols

### CODE REVIEW

**Comment by fergmastaflex** [comment]
Also includes some rubocop changes

File: lib/slide_rule/distance_calculator.rb [hunk 1]
```diff
@@ -85,10 +85,10 @@ def calculate_weighted_distances(i1, i2)
     def get_calculator(calculator)
       return calculator.new if calculator.is_a?(Class)
 
-      klass_name = "#{calculator.to_s.split('_').collect(&:capitalize).join}"
+      klass_name = calculator.to_s.split('_').collect(&:capitalize).join.to_s
       klass = begin
         ::SlideRule::DistanceCalculators.const_get(klass_name)
-      rescue(::NameError)
+      rescue::NameError
```
Comment (mattnichols) [inline lib/slide_rule/distance_calculator.rb]: Should there be a space between `rescue` and `::NameError`?

Comment (fergmastaflex) [inline lib/slide_rule/distance_calculator.rb]: Yeah rubocop did that. He's drunk and needs to go home.

File: lib/slide_rule/version.rb [hunk 2]
```diff
@@ -1,3 +1,3 @@
 module SlideRule
-  VERSION = '0.2.1'
+  VERSION = '0.2.1'.freeze
```
Comment (mattnichols) [inline lib/slide_rule/version.rb:2]: :+1:  Is that a new cop?

Comment (fergmastaflex) [inline lib/slide_rule/version.rb:2]: It is a cop, but we don't do that in our system. I added that to the ignore and meant to remove this. I can change it though.

File: lib/slide_rule/distance_calculators/day_of_month.rb [hunk 3]
```diff
@@ -8,6 +8,7 @@ class DayOfMonth
       #   Does not take into account the number of days in the actual month being considered.
       #
       def calculate(first, second)
+        return if first.nil? || second.nil?
```
Comment (mattnichols) [inline lib/slide_rule/distance_calculators/day_of_month.rb]: Should we return nil explicitly here? I prefer to do it that way if I expect a nil from the method. `return nil if first.nil? || second.nil?`

Comment (fergmastaflex) [inline lib/slide_rule/distance_calculators/day_of_month.rb]: Makes sense.

File: lib/slide_rule/version.rb (hunk 2 above)
Comment (mattnichols) [inline lib/slide_rule/version.rb:2]: I don't have a problem with this. Why not freeze it?

Comment (fergmastaflex) [inline lib/slide_rule/version.rb:2]: I don't see why not. Sounds good to me.

Comment (fergmastaflex) [inline lib/slide_rule/version.rb:2]: I removed the rubocop ignore as well.

**Comment by mattnichols** [comment]
:shipit:

**Pull request merged**